cosmo_tester/test_suites/image_based_tests/shared_resource_test.py

```diff
@@ -0,0 +1,92 @@
+# distributed under the License is distributed on an "AS IS" BASIS,
+#    * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    * See the License for the specific language governing permissions and
+#    * limitations under the License.
+
+from cosmo_tester.framework import util
+from cosmo_tester.framework.examples import get_example_deployment
+from .component_test import wait_for_execution_status
+
+
+def _infra(image_based_manager, ssh_key, logger, tenant, test_config):
+    example = get_example_deployment(
+        image_based_manager, ssh_key, logger, tenant, test_config,
+        upload_plugin=False)
+
+    example.blueprint_file = util.get_resource_path(
+        'blueprints/component/fake_infra.yaml'
+    )
+    example.blueprint_id = 'infra'
+    example.deployment_id = 'infra'
+    example.inputs = {}
+    return example
+
+
+def _app(image_based_manager, ssh_key, logger, tenant, test_config):
+    example = get_example_deployment(
+        image_based_manager, ssh_key, logger, tenant, test_config,
+        upload_plugin=False)
+
+    example.blueprint_file = util.get_resource_path(
+        'blueprints/component/shared_resource.yaml'
+    )
+    example.blueprint_id = 'app'
+    example.deployment_id = 'app'
+    example.inputs['agent_user'] = image_based_manager.username
+    example.create_secret = False   # don't try to create it twice
+    return example
+
+
+def test_shared_resource(image_based_manager, ssh_key, logger, test_config):
+    tenant = 'test_shared_resource'
+    infra = _infra(image_based_manager, ssh_key, logger, tenant, test_config)
+    app = _app(image_based_manager, ssh_key, logger, tenant, test_config)
+
+    # We're uploading a blueprint that creates an infrastructure for a VM,
+    # and then exposes capabilities, which will be used in the application
+    logger.info('Deploying the blueprint which contains the shared resource.')
+    infra.upload_blueprint()
+    infra.create_deployment()
+    logger.info('Deploying application blueprint.')
+    app.upload_and_verify_install()
+
+    with util.set_client_tenant(app.manager.client, tenant):
+        # verify both deployments are deployed
+        nodes = app.manager.client.nodes.list()
+        assert len(set(node.deployment_id for node in nodes)) == 2
+        for node in nodes:
+            if node.type == 'cloudify.nodes.Compute':
+                assert node.number_of_instances == 1
+
+        # plant runtime property which conditions the execution to fail
+        instances = app.manager.client.node_instances.list(deployment_id='app',
+                                                           node_id='app')
+        app.manager.client.node_instances.update(
+            instances[0].id,
+            runtime_properties={'fail_commit': True},
+            version=instances[0].version+1
+        )
+        rollout = app.manager.client.executions.start('app', 'rollout')
+
+        logger.info('Testing execution cancel.')
```

Comment:
This also looks very similar to the code above- maybe factor out its assertions and disabling of the trigger too, e.g. _check_rollout_cancel(app); _disable_rollout_failure(app); _check_rollout_resume(app)?